vms/proposervm/batched_vm_test.go

```diff
@@ -583,6 +588,112 @@ func TestBatchedParseBlockPreForkOnly(t *testing.T) {
 	require.Equal(builtBlk3.ID(), res[2].ID())
 }
 
+func TestBatchedParseBlockParallel(t *testing.T) {
+	require := require.New(t)
+
+	parentID := ids.ID{1}
+	timestamp := time.Unix(123, 0)
+	pChainHeight := uint64(2)
+	chainID := ids.ID{4}
+
+	vm := VM{
+		ctx: &snow.Context{ChainID: chainID},
+		ChainVM: &block.TestVM{
+			ParseBlockF: func(_ context.Context, rawBlock []byte) (snowman.Block, error) {
+				return &snowmantest.Block{BytesV: rawBlock}, nil
+			},
+		},
+	}
+
+	tlsCert, err := staking.NewTLSCert()
+	require.NoError(err)
+
+	cert, err := staking.ParseCertificate(tlsCert.Leaf.Raw)
+	require.NoError(err)
+	key := tlsCert.PrivateKey.(crypto.Signer)
+
+	makeSignedBlock := func(i int) []byte {
+		buff := make([]byte, 2)
+		binary.BigEndian.PutUint16(buff, uint16(i))
+
+		signedBlock, err := blockbuilder.Build(
+			parentID,
+			timestamp,
+			pChainHeight,
+			cert,
+			buff,
+			chainID,
+			key,
+		)
+		require.NoError(err)
+
+		return signedBlock.Bytes()
+	}
+
+	blockThatCantBeParsed := snowmantest.BuildChild(snowmantest.Genesis)
+
+	var blockBatch1 [][]byte
+	for i := 0; i < 100; i++ {
+		blockBatch1 = append(blockBatch1, makeSignedBlock(i))
+	}
+	blockBatch1[50] = blockThatCantBeParsed.Bytes()
+
+	var blockBatch2 [][]byte
+	for i := 0; i < 100; i++ {
+		blockBatch2 = append(blockBatch2, makeSignedBlock(i))
+	}
+	blockBatch2[0] = blockThatCantBeParsed.Bytes()
+
+	var blockBatch3 [][]byte
+	for i := 0; i < 100; i++ {
+		blockBatch3 = append(blockBatch3, makeSignedBlock(i))
+	}
+
+	for _, testCase := range []struct {
+		name         string
+		preForkIndex int
+		rawBlocks    [][]byte
+	}{
+		{
+			name: "empty input",
+		},
+		{
+			name:         "pre-fork is somewhere in the middle",
+			rawBlocks:    blockBatch1,
+			preForkIndex: 50,
+		},
+		{
+			name:      "first block in batch is pre fork",
+			rawBlocks: blockBatch2,
+		},
+		{
+			name:         "all blocks are post fork",
+			rawBlocks:    blockBatch3,
+			preForkIndex: 9999,
+		},
+	} {
+		t.Run(testCase.name, func(t *testing.T) {
+			blocks, err := vm.BatchedParseBlock(context.Background(), testCase.rawBlocks)
+			require.NoError(err)
+
+			var expectedBlocks [][]byte
+			for _, block := range blocks {
+				expectedBlocks = append(expectedBlocks, block.Bytes())
+			}
+
+			require.Equal(expectedBlocks, testCase.rawBlocks)
```

Comment:
This instance of `require` is bound to the outer `testing.T` so errors will be reported incorrectly and your other test cases probably won't run if one fails. The unused `t` lint failure is a false flag.

Reply:
Yep, I am sloppy :-) 

Nice catch!